### PR TITLE
Make it possible to run Proof-of-signature and Proof-of-stake in devnets/testnets

### LINKED ISSUE #1052: Error: cannot access genesis ledger account private key
When starting a seed, without propose or snarker on, I get this message.
(based on build-11856)

```
Oct 29 17:16:04 devnet100_seed_0.us-west-2 start_coda.sh[613]: (monitor.ml.Error
Oct 29 17:16:04 devnet100_seed_0.us-west-2 start_coda.sh[613]:  "cannot access genesis ledger account private key (HINT: did you forget to compile with `--profile=test`?)"
Oct 29 17:16:04 devnet100_seed_0.us-west-2 start_coda.sh[613]:  ("Raised at file \"src/error.ml\" (inlined), line 9, characters 14-30"
Oct 29 17:16:04 devnet100_seed_0.us-west-2 start_coda.sh[613]:   "Called from file \"src/option.ml\", line 69, characters 4-21"
```

Can we improve this to be a cleaner exit/fail with some more detail.
It's current developer focused, and should be improved for users too.

I'd also like to question why a seed node (not proposer or snarker) would need the genesis ledger private.

### PR DESCRIPTION
**Pull request by bkase**
Proof-of-signature and Proof-of-stake both require a private key when proposing. In this PR, we switch to demanding this private key when proposal is enabled. As such we no longer need `signer_key` for proof-of-signature.

I had to change the `Consensus_transition_data` for proof-of-signature to be a variant as we don't have the private key readily available for the genesis state. When it's genesis we just store (1,1) as the signature and ignore the signature verification iff this is the genesis transition. I believe that the SNARK circuit is still correct, but please review this part carefully.

The rest of the code just adjusts to the new lack of having this keypair available always and is relatively straightforward.

This fixes the issue we were seeing where the proposer on a devnet would die instantly (as our code was trying to read a secret key out of the genesis ledger which we no longer compile in in production-mode).

Checklist:

- [ ] Tests were added for the new behavior
- [ ] All tests pass (CI will check this if you didn't)
- [x] Does this close issues? List them:

Closes #1052

### CODE REVIEW

File: src/lib/consensus/proof_of_signature.ml [hunk 1]
```diff
@@ -1,16 +1,25 @@
+[%%import
```
Comment (bkase) [inline src/lib/consensus/proof_of_signature.ml]: Reviewer: Please look very closely at the changes in this file

File: src/lib/cli_lib/cli_lib.ml [hunk 2]
```diff
@@ -138,6 +138,7 @@ let write_keypair_exn {Keypair.private_key; public_key; _}
       Writer.close pubkey_f
   | Error e -> raise (Error.to_exn e)
 
+(** Reads a private key from [privkey_path] using [Secret_file] *)
```
Comment (cmr) [inline src/lib/cli_lib/cli_lib.ml]: did you move that or did ocamlformat?

Comment (bkase) [reply src/lib/cli_lib/cli_lib.ml]: I moved it -- it wasn't hugging the function so ocamlformat didn't know this doccomment was meant for that function

Comment (cmr) [reply src/lib/cli_lib/cli_lib.ml]: oops

**Comment by bkase** [comment]
Somewhere a SNARK assertion is failing:
```
gen_values.exe: /home/opam/app/src/_build/default/lib/snarky/src/camlsnark_c/libsnark-caml/./libsnark/reductions/r1cs_to_sap/r1cs_to_sap.tcc:348: libsnark::sap_witness<FieldT> libsnark::r1cs_to_sap_witness_map(const libsnark::r1cs_constraint_system<FieldT>&, libsnark::r1cs_primary_input<FieldT>&, libsnark::r1cs_auxiliary_input<FieldT>&, const FieldT&, const FieldT&) [with FieldT = libff::Fp_model<5l, ((const libff::bigint<5l>&)(& libff::mnt46_modulus_A))>; libsnark::r1cs_primary_input<FieldT> = std::vector<libff::Fp_model<5l, ((const libff::bigint<5l>&)(& libff::mnt46_modulus_A))>, std::allocator<libff::Fp_model<5l, ((const libff::bigint<5l>&)(& libff::mnt46_modulus_A))> > >; libsnark::r1cs_auxiliary_input<FieldT> = std::vector<libff::Fp_model<5l, ((const libff::bigint<5l>&)(& libff::mnt46_modulus_A))>, std::allocator<libff::Fp_model<5l, ((const libff::bigint<5l>&)(& libff::mnt46_modulus_A))> > >]: Assertion `cs.is_satisfied(primary_input, auxiliary_input)' failed.
```

will explore more tomorrow

**Pull request merged**